resources/api/reservation.py

```diff
@@ -768,8 +768,27 @@ def perform_create(self, serializer):
 
     def perform_update(self, serializer):
         old_instance = self.get_object()
-        new_state = serializer.validated_data.pop('state', old_instance.state)
+        old_state = old_instance.state
+        new_state = serializer.validated_data.pop("state", old_state)
         new_instance = serializer.save(modified_by=self.request.user)
+        payment_return_url = serializer.validated_data.pop("payment_return_url", None)
+        order = old_instance.get_order()
+        if (
+            order
+            and old_state == Reservation.REQUESTED
+            and new_state == Reservation.WAITING_FOR_PAYMENT
+        ):
+            if not payment_return_url:
+                raise ValidationError(_("Return URL is required to initiate the payment"))
+            else:
+                # Initiate the payment process
+                from payments.providers import get_payment_provider
```

Comment:
Does importing `get_payment_provider` cause a circular import if imported at top of module? If not, then import can be moved out of the method.

Reply:
I was quite sure it would create a circular import but seems not. I'll change this.

Reply:
Fixed